server/src/main/java/org/opensearch/action/admin/cluster/wlm/TransportQueryGroupStatsAction.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.wlm;
+
+import org.opensearch.action.FailedNodeException;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.nodes.TransportNodesAction;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.transport.TransportService;
+import org.opensearch.wlm.QueryGroupService;
+import org.opensearch.wlm.stats.QueryGroupStats;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Transport action for obtaining QueryGroupStats
+ *
+ * @opensearch.experimental
+ */
+public class TransportQueryGroupStatsAction extends TransportNodesAction<
+    QueryGroupStatsRequest,
+    QueryGroupStatsResponse,
+    TransportQueryGroupStatsAction.NodeQueryGroupStatsRequest,
+    QueryGroupStats> {
+
+    QueryGroupService queryGroupService;
+
+    @Inject
+    public TransportQueryGroupStatsAction(
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        TransportService transportService,
+        QueryGroupService queryGroupService,
+        ActionFilters actionFilters
+    ) {
+        super(
+            QueryGroupStatsAction.NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            QueryGroupStatsRequest::new,
+            NodeQueryGroupStatsRequest::new,
+            ThreadPool.Names.MANAGEMENT,
+            QueryGroupStats.class
+        );
+        this.queryGroupService = queryGroupService;
+    }
+
+    @Override
+    protected QueryGroupStatsResponse newResponse(
+        QueryGroupStatsRequest request,
+        List<QueryGroupStats> queryGroupStats,
+        List<FailedNodeException> failures
+    ) {
+        return new QueryGroupStatsResponse(clusterService.getClusterName(), queryGroupStats, failures);
+    }
+
+    @Override
+    protected NodeQueryGroupStatsRequest newNodeRequest(QueryGroupStatsRequest request) {
+        return new NodeQueryGroupStatsRequest(request);
+    }
+
+    @Override
+    protected QueryGroupStats newNodeResponse(StreamInput in) throws IOException {
+        return new QueryGroupStats(in);
+    }
+
+    @Override
+    protected QueryGroupStats nodeOperation(NodeQueryGroupStatsRequest nodeQueryGroupStatsRequest) {
+        return queryGroupService.nodeStats();
+    }
+
+    /**
+     * Inner QueryGroupStatsRequest
+     *
+     * @opensearch.experimental
+     */
+    public static class NodeQueryGroupStatsRequest extends TransportRequest {
+
```

Comment:
Do we really need this as this wrapper is not doing anything additional and the underlying object type is already `TransportRequest` ?

Reply:
I'm following the structure of `TransportNodesStatsAction`. `NodeQueryGroupStatsRequest` is used as a per-node action handler inside Transport Action, and QueryGroupStatsRequest represents the action for all nodes as a whole. The `NodeQueryGroupStatsRequest` serves as the container for the request sent to each node, allowing the framework to serialize and execute that request on the individual nodes.